### PR TITLE
feat: Lens runtime config

### LINKED ISSUE #2496: Lens runtime db.Option
Allow the user to specify a custom lens runtime for a db instance.

### PR DESCRIPTION
**Pull request by nasdf**
## Relevant issue(s)

Resolves #2496 

## Description

This PR adds a lens runtime option. It also includes some cleanup and additional tests.

## Tasks

- [x] I made sure the code is well commented, particularly hard-to-understand areas.
- [x] I made sure the repository-held documentation is changed accordingly.
- [x] I made sure the pull request title adheres to the conventional commit style (the subset used in the project can be found in [tools/configs/chglog/config.yml](tools/configs/chglog/config.yml)).
- [x] I made sure to discuss its limitations such as threats to validity, vulnerability to mistake and misuse, robustness to invalidation of assumptions, resource requirements, ...

## How has this been tested?

`make test`

Specify the platform(s) on which this was tested:
- MacOS

### CODE REVIEW

File: db/config.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+const (
+	defaultMaxTxnRetries  = 5
+	updateEventBufferSize = 100
+)
+
+// Option is a funtion that sets a config value on the db.
+type Option func(*db)
+
+// WithACP enables access control. If path is empty then acp runs in-memory.
+func WithACP(path string) Option {
```
Comment (AndrewSisley) [inline db/config.go:32]: thought: Seeing these 'new' ACP funcs was a surprise to me when reviewing, and there is not really any documentation in the PR description, and it is done in the same commit as the new Lens stuff.  This forces reviewers to spend more time/energy figuring out what is existing stuff being moved, and what is new stuff being added.  I suggest (in the future, not asking to to change this now) separate commits for this kind of stuff, and please be a little more explicit in the PR description as to what has been done.

File: lens/config.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package lens
+
+import "github.com/lens-vm/lens/host-go/engine/module"
+
+// Option is a funtion that sets a config value on the lens registry.
+type Option func(*lensRegistry)
```
Comment (AndrewSisley) [inline lens/config.go:16]: todo: Please do not define public-public stuff (i.e. not just internal-public stuff) in the lens package.  Users (or our integration tests) should not ever be referencing this package, and by adding stuff that requires the direct reference you are making the existing internal-public stuff public-public.

Please move these somewhere else, or given that they are setting private stuff, maybe have public-public funcs in `client` or `db` that call these internal-public functions, and make sure you remove the reference to this package from the integration tests.

Comment (nasdf) [reply lens/config.go:16]: We should move the lens package into an `internal` sub-package if it's meant to be internal only.

Comment (AndrewSisley) [reply lens/config.go:16]: Yes, this has been a long running battle, not everyone wants an internal package, although as far as I know everyone agrees that only a couple of packages are public.  I created an issue for a proper internal package and we can discuss it in the standup - this conversation seems to happen every other month or so.

https://github.com/sourcenetwork/defradb/issues/2507

Comment (AndrewSisley) [reply lens/config.go:16]: To clarify, as far as I know, the fact that only a couple of packages are public is not really disputed by anyone - we all agree that public stuff can only live in `client` etc.  What is disputed is whether that public/internal split should be enforced/documented by an `internal` directory or not.

File: lens/registry.go [hunk 3]
```diff
@@ -84,24 +84,30 @@ const DefaultPoolSize int = 5
 // NewRegistry instantiates a new registery.
 //
 // It will be of size 5 (per schema version) if a size is not provided.
-func NewRegistry(lensPoolSize immutable.Option[int], db TxnSource) client.LensRegistry {
-	var size int
-	if lensPoolSize.HasValue() {
-		size = lensPoolSize.Value()
-	} else {
-		size = DefaultPoolSize
+func NewRegistry(
```
Comment (AndrewSisley) [inline lens/registry.go:87]: praise: The new param order makes a lot more sense, thanks for making `db` the first one :)

File: db/db.go [hunk 4]
```diff
@@ -149,15 +107,12 @@ func newDB(
 
 	// apply options
 	for _, opt := range options {
-		if opt == nil {
```
Comment (AndrewSisley) [inline db/db.go:152]: praise: Thanks for removing this :)

**Review by AndrewSisley — APPROVED** [review]
LGTM, thanks Keenan, and sorry about the hassle/confusion RE lens/internal :)

**Pull request merged**